Fixed interval/stop to work with ramping concurrency

jberkhahn opened the pull request:
Changed runner.go so that a new schedule is created for each separate ExecuteConcurrently(). The changes to benchmarker.go are due to gofmt.

jberkhahn committed:
Fixed interval/stop to work with ramping concurrency

julz commented:
looks mostly good, but you have the 'go fmt' changes and the new code in one commit, which means I can't easily tell if you actually added a test for the fix.. if I revert the changes in runner.go, will runner_test.go fail?

jberkhahn commented:
Yes, the tests will fail. I didn't actually add any tests, merely  changed existing ones, as the fix is a glorified private wrapper function.

julz commented:
I see.. they won't compile at all if I revert runner.go, but if I add a `Start()` method that just returns the existing channel (and otherwise keep the implementation the same as it was) they won't fail. Let's add a test that shows that if you call `.Start()` on a concurrencySchedule it starts again from 0 (which is the behaviour we're relying on).

jberkhahn commented:
Ok, changed start to be a private method as there's really no reason for it to be exported, and added a test that checks that calling start repeatedly on a schedule restarts it.

julz commented on experiment/runner_test.go:
```diff
@@ -297,10 +299,42 @@ var _ = Describe("ExperimentConfiguration and Sampler", func() {
+				It("creates a new schedule each time start() is called", func() {
+					scheduler := linearSchedule(1, 3, 3*time.Second)
+					schedule := scheduler.start()
+					for i := 0; i < 1; i++ {
```
I might be being dumb.. but what's the point of this loop? ditto the loop on 325..

jberkhahn commented:
Woops, that was a remnant of an earlier version of the test. Removed.
